### DISCUSSION TITLE
`Align2` represented as a 2-element array instead of a tuple?

### DISCUSSION THREAD

**Discussion by Mingun** [emilk/egui#1053] [Q&A]
It is a bit strange and probably error-prone, because you can index `Align2` type [by any number](https://github.com/emilk/egui/blob/de1a1ba9b2f1b19a35a1aae419725ca30e9193a4/emath/src/align.rs#L175-L189) which could cause panic. Isn't it would be better to use 2-tuple representation. Or, even better, just an ordinal `struct { pub x, pub y }`? I see no benefits from the current array representation.

**Comment by emilk** [comment]
`struct Align2 { pub x: Align, pub y: Align }` would make sense imho, and would make it similar to `Pos2` and `Vec2`.
